Quickstart/Sample/Program.cs

```diff
@@ -1,20 +1,50 @@
-﻿using Microsoft.AspNetCore.Hosting;
+﻿using Auth0.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using SampleMvcApp.Support;
+using System.Net;
 
-namespace SampleMvcApp
+using Microsoft.IdentityModel.Logging;
+
+var builder = WebApplication.CreateBuilder(args);
+
+//Adding these 2 statements prevents .NET 6 from advising you we must declare these in the old ConfigureServices
+//method from previous .NET versions.
+builder.Services.AddAuthorization();
+builder.Services.AddControllers();
+//To use MVC we have to explicitly declare we are using it. Doing so will prevent a System.InvalidOperationException.
+builder.Services.AddMvc();
+builder.Services.AddControllersWithViews();
+builder.Services.AddAuth0WebAppAuthentication(options =>
+{
+    options.Domain = builder.Configuration["Auth0:Domain"];
+    options.ClientId = builder.Configuration["Auth0:ClientId"];
+});
+
+// Configure the HTTP request pipeline.
+builder.Services.ConfigureSameSiteNoneCookies();
+var app = builder.Build();
+
+if (!app.Environment.IsDevelopment())
 {
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
-
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.UseStartup<Startup>();
-                });
-    }
+    app.UseExceptionHandler("/Home/Error");
+    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+    app.UseHsts();
 }
+
+app.UseHttpsRedirection();
```

Comment:
Any reason for adding this while we didn't have this before?

Reply:
No real reason for that. I just used it because that was what was in the [MS ASP.NET 6 code samples](https://learn.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-7.0&tabs=visual-studio#:~:text=hsts.%0A%20%20%20%20app.UseHsts()%3B%0A%7D-,app.UseHttpsRedirection()%3B,-app.UseStaticFiles()%3B%0A%0Aapp) from migrating to the minimal hosting model. I'm getting rid of that. Taking it out on my end resulted in no changes.